### DISCUSSION TITLE
Showing a tooltip for a disabled items

### DISCUSSION THREAD

**Discussion by marcospassos** [ariakit/ariakit#1410] [Questions] [answered]
Ariakit uses `pointer-events` as a bullet-proof solution for disabling elements. It, however, prevents legitimate use cases like showing a tooltip to help the user understand why a select box option is disabled, for instance. One can always ignore the `pointer-events`, but the option would still become `active` in this example.

What's the recommendation in such cases?

**Comment by diegohaz** [comment, marked as answer]
You can use the `accessibleWhenDisabled` prop:
```jsx
<TooltipAnchor as={Button} disabled accessibleWhenDisabled />
```

I'd recommend letting it be accessed with a mouse/keyboard (which is what the `accessibleWhenDisabled` prop does).

You can just overwrite `pointerEvents` and style `[data-active-item][aria-disabled]` differently if you really don't want that:

```jsx
<Item disabled style={{ pointerEvents: "initial" }} />
```

**Reply by marcospassos** [reply]
But it doesn't play well with abstractions:

```tsx
export const Tooltip: FunctionComponent<TooltipProps> = ({children, message}) => {
    const tooltip = useTooltipState({
        placement: "right",
    });

    return (
        <>
            <TooltipAnchor state={tooltip}>
                {anchorProps => cloneElement(children, anchorProps)}
            </TooltipAnchor>
            <BaseTooltip state={tooltip} className="tooltip">
                <TooltipArrow state={tooltip} className="tooltip-arrow" />
                {message}
            </BaseTooltip>
        </>
    );
}
```

**Reply by diegohaz** [reply]
You would pass `accessibleWhenDisabled` to the same component you pass the `disabled` prop.

**Reply by marcospassos** [reply]
I tried, but it doesn't seem to work:

```tsx

export const CustomSelectOption = forwardRef<HTMLDivElement, ComboboxOptionProps>(
    (props, ref) => {
        const {children, ...htmlProps} = props;

        return (
            <Button as="div" accessibleWhenDisabled {...htmlProps} ref={ref}>
                Label
            </Button>
        );
    }
);
```

The `pointer-events` inhibits the hover event to fire. Removing it makes the hover on the select option to work even disabled.

**Reply by diegohaz** [reply]
What's the component that's receiving the `disabled` prop?

**Reply by marcospassos** [reply]
That same button. If I omit the `pointer-evens` and disable the `focusOnHover` on the select item, I can get it working by using `:hover` instead of `[data-active-item]`. But I don't fully understand the consequences of disabling it for accessibility. Is it a huge penalty?

**Reply by diegohaz** [reply]
It's hard to say without seeing the UI in action. But that way I don't think keyboard users would have access to the tooltip as mouse users would.

I recommend the `accessibleWhenDisabled` approach and `[data-active-item]` (and not `:hover`). You can style it differently with the `[data-active-item][aria-disabled]` selector.

The `accessibleWhenDisabled` prop should be passed to the *same* component where the `disabled` prop is passed. If you have `<SelectItem disabled>` anywhere, the prop should be there, not in the underlying `<Button>`. I'm emphasizing this again because your `CustomSelectOption` looks like a generic button that may receive props from another Ariakit component (like `SelectItem`).

**Reply by marcospassos** [reply]
You're right. That's what I was doing. Passing it to the `<SelectItem/>` prevents it from passing the `pointer-events` downstream. I thought the active status was a side-effect, but now I see it behaves like a focus state, so it seems ok.

Thanks for the help!
